### PR TITLE
Music Player now uses Qtmultimedia instead of libmpg123

### PR DESCRIPTION
**Pull request by ThomasVy**
Replaced libmpg123 with Qtmultimedia
Cleaned files
 - Deleted unused files
 - forward declared classes in headers instead of including them
 - changed some enums to be enum class (stronger typed)
 - changed old C-casts to C++ casts (eg. static_cast<int>(blah))
 - fix some other warnings

### CODE REVIEW

File: src/BusinessLayer/SongControl/SongControl.h [hunk 1]
```diff
@@ -1,19 +1,10 @@
...
-#include <stdio.h>
-#include <stdlib.h>
-#include <string>
-#include <vector>
 #include <QString>
 #include <QTime>
 #include <QVector>
-#include <QFileInfo>
+
+#pragma once
```
Comment (bill-luu) [inline src/BusinessLayer/SongControl/SongControl.h]: This should still be the first line

File: src/BusinessLayer/SongPlayer/SongPlayer.cpp [hunk 2]
```diff
@@ -182,19 +192,23 @@ void SongPlayer::toggleLoop()
...
+    album_ = mediaPlayer_->metaData(QMediaMetaData::AlbumTitle).toString();
 
     //remove all spaces in album name for easier access to file path of album
     album_.replace(" ", "");
 
     cover_ = (ALBUM_FILE_PATH);
     cover_.append(album_);
 
-    QPixmap img(cover_);
+//    QPixmap img(cover_);
```
Comment (bill-luu) [inline src/BusinessLayer/SongPlayer/SongPlayer.cpp]: delete this

File: src/ViewLayer/ProgressBar/ProgressBar.h [hunk 3]
```diff
@@ -1,9 +1,10 @@
 #pragma once
 
 #include <QWidget>
-#include <QPaintEvent>
 #include <QPainter>
```
Comment (bill-luu) [inline src/ViewLayer/ProgressBar/ProgressBar.h]: See if this can also be forward declared.

File: src/ViewLayer/SongPlayerView/SongPlayerView.cpp [hunk 4]
```diff
@@ -1,5 +1,17 @@
 #include <QDir>
 #include "SongPlayerView.h"
+#include "SongPlayerUI/I_SongPlayerUi.h"
+#include <QLabel>
+#include <QPushButton>
+#include <QObject>
+#include <QLabel>
+#include <QFuture>
+#include "../BusinessLayer/SongPlayer/SongPlayer.h"
```
Comment (bill-luu) [inline src/ViewLayer/SongPlayerView/SongPlayerView.cpp]: Group the includes to be 
```
<QT includes>
"Our files"
```

File: src/ViewLayer/ViewContainer.cpp [hunk 5]
```diff
@@ -1,4 +1,8 @@
 #include "ViewContainer.h"
+#include "../BusinessLayer/BusinessContainer.h"
+#include <SongPlayerUI/SongPlayerUi.h>
+#include <ProgressBar/ProgressBar.h>
+#include <SongPlayerView/SongPlayerView.h>
```
Comment (bill-luu) [inline src/ViewLayer/ViewContainer.cpp]: Include grouping

**Review by bill-luu — CHANGES_REQUESTED** [review]

File: src/BusinessLayer/SongControl/SongControl.h (hunk 1 above)
Comment (ThomasVy) [reply src/BusinessLayer/SongControl/SongControl.h]: Ok

File: src/BusinessLayer/SongPlayer/SongPlayer.cpp (hunk 2 above)
Comment (ThomasVy) [reply src/BusinessLayer/SongPlayer/SongPlayer.cpp]: Done

File: src/ViewLayer/ProgressBar/ProgressBar.h (hunk 3 above)
Comment (ThomasVy) [reply src/ViewLayer/ProgressBar/ProgressBar.h]: Can use #include <QPen>

File: src/ViewLayer/SongPlayerView/SongPlayerView.cpp (hunk 4 above)
Comment (ThomasVy) [reply src/ViewLayer/SongPlayerView/SongPlayerView.cpp]: Good

File: src/ViewLayer/ViewContainer.cpp (hunk 5 above)
Comment (ThomasVy) [reply src/ViewLayer/ViewContainer.cpp]: Done

File: src/BusinessLayer/SongControl/SongControl.cpp [hunk 6]
```diff
@@ -1,13 +1,14 @@
 #include <QDir>
-#include "SongControl.h"
+#include <dirent.h>
```
Comment (bill-luu) [inline src/BusinessLayer/SongControl/SongControl.cpp:2]: Are we able to drop this now?

File: src/ViewLayer/FontLoader/FontLoader.h [hunk 7]
```diff
@@ -1,6 +1,5 @@
 #pragma once
 
-#include <QObject>
 #include <QHash>
 #include <QFont>
```
Comment (bill-luu) [inline src/ViewLayer/FontLoader/FontLoader.h]: Can QFont be forward declared?

**Review by bill-luu — CHANGES_REQUESTED** [review]

**Comment by bill-luu** [comment]
Can we also update the download script? Since it now no longer needs to use eyeD3 to get the song images

File: src/BusinessLayer/SongControl/SongControl.cpp (hunk 6 above)
Comment (ThomasVy) [reply src/BusinessLayer/SongControl/SongControl.cpp:2]: <dirent.h>? I didn't remove the usage of dirent so we can't remove it yet. I'll leave that as a future bug/refactor.

File: src/ViewLayer/FontLoader/FontLoader.h (hunk 7 above)
Comment (ThomasVy) [reply src/ViewLayer/FontLoader/FontLoader.h]: Good eye. Did good

**Comment by ThomasVy** [comment]
> Can we also update the download script? Since it now no longer needs to use eyeD3 to get the song images

The script never worked in the first place, so I rather just leave it as it is for now until we think of a better idea of downloading the files.

**Review by navjotbrar — APPROVED** [review]
Looks good to me, just fix what Bill mentioned.

**Review by bill-luu — APPROVED** [review]

File: src/BusinessLayer/SongControl/SongControl.cpp (hunk 6 above)
Comment (bill-luu) [reply src/BusinessLayer/SongControl/SongControl.cpp:2]: Make a new issue for it!

**Comment by bill-luu** [comment]
Make a new issue for the script stuff

**Pull request merged**